### PR TITLE
Migrate RemoteMediaServer to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Migrates the RemoteMediaServer.java file to kotlin following the migration guide.

## Fixes

Closes #10145 

## How Has This Been Tested?

Ran the usual tests.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        return defaultAnkiWebUrl
+    }
+
+    @Throws(UnknownHttpResponseException::class, MediaSyncException::class)
+    fun begin(): JSONObject {
+        return try {
+            mPostVars = HashMapInit(2)
+            mPostVars["k"] = mHKey
+            mPostVars["v"] =
+                String.format(Locale.US, "ankidroid,%s,%s", pkgVersionName, Utils.platDesc())
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt:68]:
```suggestion
            mPostVars["v"] = String.format(Locale.US, "ankidroid,%s,%s", pkgVersionName, Utils.platDesc())
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    @Throws(UnknownHttpResponseException::class)
+    fun downloadFiles(top: List<String?>?): ZipFile {
+        var resp: Response? = null
+        return try {
+            resp = super.req(
+                "downloadFiles",
+                getInputStream(Utils.jsonToString(JSONObject().put("files", JSONArray(top))))
+            )
+            val zipPath =
+                col!!.path.replaceFirst("collection\\.anki2$".toRegex(), "tmpSyncFromServer.zip")
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt:110]:
```suggestion
            val zipPath = col!!.path.replaceFirst("collection\\.anki2$".toRegex(), "tmpSyncFromServer.zip")
```

**Review by Akshay0701 — COMMENTED** [review]
Thanks!
looks good to me, one minor chnage

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt (hunk 2 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt]: Done!

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt (hunk 1 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt]: Done!

**Review by Akshay0701 — APPROVED** [review]

**Pull request merged**